cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -70,12 +74,54 @@ struct sample_edges_op_t {
   }
 };
 
+template <typename vertex_t, typename bias_t>
+struct biased_random_walk_e_bias_op_t {
+
+  __device__ bias_t
+  operator()(vertex_t, vertex_t, bias_t src_out_weight_sum, thrust::nullopt_t, bias_t weight) const
+  {
+    return weight / src_out_weight_sum;
+  }
+};
+
+template <typename vertex_t, typename weight_t>
+struct biased_sample_edges_op_t {
+  __device__ thrust::tuple<vertex_t, weight_t>
+  operator()(vertex_t, vertex_t dst, weight_t, thrust::nullopt_t, weight_t weight) const
+  {
+    return thrust::make_tuple(dst, weight);
+  }
+};
+
+template <typename vertex_t, typename weight_t>
+struct node2vec_sample_edges_op_t {
```

Comment:
Do you plan to implement both biased & node2vec random walks in a single PR?

Reply:
Yes, I was instructed to just make them a single PR

Reply:
We could split them up if that's easier.  Smaller pull requests are easier to review... but if the code changes are pretty small it wouldn't be too bad to have them together.